nucleus/common/common-util/src/main/java/com/sun/enterprise/security/store/AsadminSecurityUtil.java

```diff
@@ -206,16 +207,24 @@ private char[] promptForPassword() {
      * @return the AsadminTruststore object
      */
     public AsadminTruststore getAsadminTruststore() {
-        return asadminTruststore;
+        try {
+            return openTruststore(password);
+        } catch (CertificateException | KeyStoreException | NoSuchAlgorithmException | IOException e) {
+            throw new RuntimeException(e);
```

Comment:
If you catch all specific RuntimeExceptions to just replace with a generic it make it harder for a caller to distinguish to different exceptions.

Reply:
Sorry. My Bad. None of them are. But wouldn't it be better to add some message i.e. in this case 'Cannot open truststore.'